kod/object/active/holder/room.kod

```diff
@@ -5127,6 +5136,26 @@ messages:
    {
       return $;
    }
+   
+   StartTemporaryNoPK()
+   {
+      if NOT (piRoom_flags & ROOM_NO_PK)
+      {
+         piRoom_flags = piRoom_flags | ROOM_NO_PK;
+         ptNoPKTimer = CreateTimer(self,@EndTemporaryNoPK,60000*60);
+      }
+      return;
+   }
+   
+   EndTemporaryNoPK()
+   {
+      ptNoPKTimer = $;
+      if NOT (piRoom_flags & ROOM_NO_PK)
```

Comment:
This message should just call Send(self,@SetRoomFlagToDefault,#flag=ROOM_NO_PK); as the way you have it currently it won't reset the flag.

Also to make it so we can call this message manually, add a timer parameter to EndTemporaryNoPK (default $) and then delete ptNoPKTimer if timer is $.